subjects/localhost/README.md

```diff
@@ -0,0 +1,162 @@
+
+### Instructions
+
+#### Server
+
+The project can be written in one of these languages [Go,C++,C...]
+
+```console
+$ ./localhost [CONFIG FILE]
+Lisning on  [IP]:[PORT]
+Lisning on  [IP]:[PORT2]
+```
+- Your server must not crash for any reason.
+- Any request to your server should never hang forever.
+- Your server can listen on multiple servers and ports at the same time without any conflict 
+- You must use the single process and no treads.
+- You must use `exec` functions only in `CGI` part
+- Your server must be compatible with the last version of your chosen browser 
+- You can use `NGINX` to compare headers and answer behaviors with your server
+- You must manage at least [`GET`, `POST`, `DELETE`] methods.
+- You must handle customization errors **(Binding error, default page error...)**
+- You must create error page at least for [400,311,403,404,405,413]
+- You must execute CGI based on certain file extensions such us [`.php`,`.py`,...]
+- The `CGI` should run in the correct directory for relative path file access.
+- You must use the enverement paramatre of cgi. 
+- Your server must receive a request from the browser and send a response using the HTTP header and body
+- Your server must call `select` function (or equivalent function) one time only
+- You must manage chunked and unchunked request.
+- You must set the right status for each response
+
+here is an example of an HTTP request with `GET` method
+
+```http
+GET /Welcome.html HTTP/1.1
+User-Agent: Mozilla/4.0 (compatible; MSIE5.01; Windows NT)
+Host: www.01talent.com
+Accept-Language: en-us
+Accept-Encoding: gzip, deflate
+Connection: Keep-Alive
+```
+and here is an example of an HTTP response : 
+
+```http
+HTTP/1.1 404 Not Found
+Date: Thu, 19 Jul 2022 10:36:20 GMT
+Server: Hserver/1.1.0
+```
+> Stress tests your server. It must stay available at all cost. use this command to test it  `siege -b [IP]:[PORT]`
+	the availability should be up to 95.99
+	
+> Learn about  [CGI](https://en.wikipedia.org/wiki/Common_Gateway_Interface)
+
+
+#### Configurations File
+
+This an example of a simple server configuration.
+```
+server {
+    listen                          80;
+    
+    server_name                     localhost;
+	
+    root                            /usr/share/nginx/html;
+    
+    error_page          Mozilla 404        /usr/share/nginx/errors/404.html;
+    
+    client_body_size                10m; 
+    location /
+    {
+        http_methods  GET  POST DELETE;
+        
+        index index.html
+        
+        upload_status on; //if it's "on," you should add the "upload_path"; otherwise, if it's "off," there is no need for "upload_status." 
+        
+        upload_path         /usr/share/nginx/upload; //depends on "upload_status"
+    }
+}
+```
+
+You have to base your server on this example but You have tested more complex configurations with multiple locations and ports and multiple servers such as :
```

Comment:
"but you must test also more complex configurations with multiple locations and ports and multiple servers such as:"